python/ray/data/datasource/datasink.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Iterable, List
+
+from ray.data._internal.execution.interfaces import TaskContext
+from ray.data.block import Block
+
+WriteResult = Any
+
+
+class Datasink:
+    def on_write_start(self, **_) -> None:
```

Comment:
Should we allow passing an args context here?

Reply:
I'm not sure if I understand. Could you elaborate?

Reply:
Oh I was thinking if there were args that the user wanted to pass to `on_write_start` (or the other methods). Currently this interface doesn't seem to allow that.

Reply:
You'd define the parameters through the constructor and access them as attributes. My thinking is that'd be more explicit than keyword-arguments or an arbitrary context.

Reply:
Okay, sounds good.